Visitor can Register for an account

DIzosimov opened the pull request:
[PT Link](https://www.pivotaltracker.com/story/show/169285628)

## PR Content
- Signup Form Component
- Redux Token Auth implementation for user registration
- State/Event Handlers
- Styling

## What have we learned
- How to implement redux token auth
- How to use debuggers and logger
- The use of props

## Screenshots
<img width="1155" alt="Screenshot 2019-10-28 at 13 39 44" src="https://user-images.githubusercontent.com/53824850/67682636-1c7be800-f990-11e9-95f9-2b46040785d8.png">

yasmineezequiel committed:
Adds react.

yasmineezequiel committed:
Resolves merge conflicts.

yasmineezequiel committed:
Adds article list unit test.

beccaburns committed:
Creates ListArticle component.

yasmineezequiel committed:
Adds integration file and User can view listed articles spec for unit testing.

Commit:
Create ListaArticles component & add to App.js

yasmineezequiel committed:
Mocks backend response of articles listed, creates json articles file.

Commit:
Add methods to ListArticles component

DIzosimov committed:
Merge branch 'list_article' of https://github.com/shsteindorsson/classy_news_client into list_article

DIzosimov committed:
Changes requestArticle module, removes redundant constructors, adds if else statement in ListArticle component to alter state of renderArticleList

Commit:
Add error messages to ListArticles component

Set url for RequestArticles module.

Commit:
Add semantic elements to component and add to feature test

DIzosimov committed:
Adds sad path testing for listArticle component

yasmineezequiel committed:
Adds integration UserCanSignUp for unit testing.

beccaburns committed:
Adds inputs for user signup form. Creates semantic ui form in app.js for user signup.

Commit:
Move SignupForm over to seperate component

DIzosimov committed:
Merge branch 'user_registration' of https://github.com/shsteindorsson/classy_news_client into user_registration

DIzosimov committed:
Adds signUpHandler to app.js, wraps App component with redux functionality, adds reduxTokenAuthConfig, adds rootReducer, adds store configuration

lucamarial committed:
Implements redux registerUser functionality

Adds event handlers for registerUser functionality.

Commit:
Install Redux & add id's to form elements

Commit:
Remove semicolons and debuggers

DIzosimov committed:
Adds registerUser attributes in reduxTokenAuthConfig, adds welcome message and creates an if else statement for signedIn vs nonSignedIn users.

DIzosimov committed:
Adds semantic to index.js, adds cancel button for signup form, adds styling to select element in signupform with css, adds general styling with semantic to signup form.

DIzosimov committed:
Removes unecessary line of code

beccaburns committed:
Stubs user information for registration welcome message.

DIzosimov committed:
Merge branch 'development' of https://github.com/CraftAcademy/classy_news_client into user_registration

DIzosimov committed:
Adds development to branch and removes unecessary spec file.

Commit:
Add link to Semantic-ui-css to index

yasmineezequiel committed:
Adds image to list of articles in landing page.

Commit:
Add Semantic styling to article with misplaced image

beccaburns committed:
Merge branch 'list_article' of https://github.com/shsteindorsson/classy_news_client into list_article

faraznaeem commented on cypress/integration/userCanSignUp.spec.js:
```diff
@@ -0,0 +1,32 @@
+    })
+    cy.get('#submit-signup-form').click()
+    cy.get('#welcome-message').should('contain', 'Hello name')
+  })
```
You are missing a sad path for this feature

DIzosimov replied:
fixed :)

beccaburns committed:
Adds Item Group Semantic UI for List Articles.

beccaburns committed:
Updates font for platform styling.

Commit:
Add ingress method to ListArticles component, change matchers for feature test

Remove redundant component file

faraznaeem committed:
Merge pull request #7 from shsteindorsson/list_article

User* can view all listed articles

DIzosimov committed:
-Adds sad path fixture file for user registration.
-Adds sad path integration test for user registration.
-Adds error message display in App.js

piavW committed:
Merges with upstream

piavW committed:
Refactors sign-up related code into it's own component.
